src/content_scripts/applyColorSchemeJs.js

```diff
@@ -4,14 +4,47 @@
 
 "use strict";
 
-const overwroteMatchMedia = false;
+let overwroteMatchMedia = false;
 
-const ADDON_FAKED_WARNING = "matchMedia has been faked/overwritten by add-on website-dark-mode-switcher; see https://github.com/rugk/website-dark-mode-switcher/. If it causes any problems, please open an issue.";
+const ADDON_FAKED_WARNING = "MediaQueryList has been faked by add-on website-dark-mode-switcher; see https://github.com/rugk/website-dark-mode-switcher/. If it causes any problems, please open an issue.";
+let loggedFakedWarning = false;
 
-/* globals COLOR_STATUS, MEDIA_QUERY_COLOR_SCHEME, MEDIA_QUERY_PREFER_COLOR, fakedColorStatus, getSystemMediaStatus */
+// instances of `MediaQueryList`s with listeners
+const setMediaQueryLists = new Set();
+
+// func -> { hook, setMediaQueryList, setMediaQueryListOnChange }
+const weakmapFuncToEntry = new WeakMap();
+
+// hook -> func
+const weakmapHookToFunc = new WeakMap();
+
+const privilegedOnChangeGetter = Reflect.getOwnPropertyDescriptor(MediaQueryList.prototype, 'onchange').get;
+
+const MediaQueryListPrototype = MediaQueryList.prototype.wrappedJSObject;
+const originalAddListener = MediaQueryListPrototype.addListener;
+const originalRemoveListener = MediaQueryListPrototype.removeListener;
+const originalMatchesGetter = Reflect.getOwnPropertyDescriptor(MediaQueryListPrototype, 'matches').get;
+const originalOnChangeGetter = Reflect.getOwnPropertyDescriptor(MediaQueryListPrototype, 'onchange').get;
+const originalOnChangeSetter = Reflect.getOwnPropertyDescriptor(MediaQueryListPrototype, 'onchange').set;
+
+const EventTargetPrototype = window.EventTarget.prototype.wrappedJSObject;
+const originalAddEventListener = EventTargetPrototype.addEventListener;
+const originalRemoveEventListener = EventTargetPrototype.removeEventListener;
+
+// ugly juggling principals
+const unsafeObjectCreate = window.wrappedJSObject.Object.create;
+
+// Whether we are dispatching "change" events
+let dispatching = false;
+
+/* globals COLOR_STATUS, MEDIA_QUERY_COLOR_SCHEME, MEDIA_QUERY_PREFER_COLOR, fakedColorStatus, getSystemMediaStatus, jsLastColorStatus */
 
 // eslint does not include X-Ray vision functions, see https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/Sharing_objects_with_page_scripts
-/* globals exportFunction, cloneInto */
+/* globals exportFunction */
+
+//
+// See https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList
```

Comment:
That is not JSDoc syntax.

Altghough we don't use CommonJS modules, this site here shows how a header should be formatted that applies to the complete file: https://jsdoc.app/howto-commonjs-modules.html